### ISSUE TITLE
Desktop notifications

### ISSUE THREAD

**Issue by TheJaredWilcurt** [tauri-apps/tauri#155]
**Describe the solution you'd like**

An easy API to interact with whatever the current OS's desktop notifications system is.

**Describe alternatives you've considered**

Manually launching a new window disguised to look like a desktop notification (not worth the effort)

**Comment by nklayman** [comment]
Would we want to "polyfill" the web notification api so that notifications work the same as in a browser?

**Comment by nothingismagick** [comment]
We didn't have this on our roadmap, but I like @nklayman 's idea for polyfilling. Nice one.

**Comment by nothingismagick** [comment]
Seems like this would be a good library to use for that (behind a feature flag of course):
https://docs.rs/notify/5.0.0-pre.1/notify/

I think the right approach would be to make a tauri API that hooks into this feature that can then be called from JS or from RS

**Comment by nothingismagick** [comment]
here's the branch where this is happening:
https://github.com/tauri-apps/tauri/tree/feature/notification

**Comment by nothingismagick** [comment]
And here's the PR
https://github.com/tauri-apps/tauri/pull/505

**Comment by lucasfernog** [comment]
Closed by #505 
The notifications API is going to be accessible with the Notification WEB API. Currently we'll support `title`, `body` and `icon`.

Example:
```js
Notification.requestPermission()
  .then(response => {
    if (response === 'granted') {
      new Notification('title', { body: 'some text' })
    }
  })
```
